src/services/organizationService.ts

```diff
@@ -490,6 +499,96 @@ export async function registerOrganization(
   }
 }
 
+export function addDelegate(
+  ipaCode: string,
+  userEmail: string
+): TaskEither<
+  // tslint:disable-next-line:max-union-size
+  IResponseErrorInternal | IResponseErrorNotFound | IResponseErrorConflict,
+  IResponseSuccessCreation<Organization>
+> {
+  const genericErrorHandler = (error: unknown) => {
+    log.error("An error occurred adding the delegate. %s", error);
+    return ResponseErrorInternal("An error occurred adding the delegate.");
+  };
+  return tryCatch<
+    IResponseErrorInternal | IResponseErrorNotFound | IResponseErrorConflict,
+    OrganizationModel
+  >(
+    () =>
+      OrganizationModel.findOne({
+        where: { ipaCode }
+      }),
+    genericErrorHandler
+  )
+    .chain(
+      fromPredicate(
+        _ => _ !== null,
+        () =>
+          ResponseErrorNotFound(
+            "Not found",
+            "Registered organization does not exists"
+          )
+      )
+    )
+    .chain(
+      fromPredicate(
+        _ =>
+          _.registrationStatus ===
+          OrganizationRegistrationStatusEnum.REGISTERED,
+        () =>
+          ResponseErrorConflict(
+            "The organization has not completed the registration process yet."
+          )
+      )
+    )
+    .chain(organizationModel =>
+      tryCatch<IResponseErrorInternal, OrganizationModel>(
+        () =>
+          OrganizationUserModel.create({
+            createdAt: Date.now(),
+            organizationIpaCode: organizationModel.ipaCode,
+            updatedAt: Date.now(),
+            userEmail,
+            userRole: UserRoleEnum.ORG_DELEGATE
+          }).then(_ => organizationModel),
+        genericErrorHandler
+      )
+    )
+    .chain(organizationModel =>
+      tryCatch(
+        () =>
+          organizationModel.reload({
+            include: [
+              {
+                as: "users",
+                model: User
+              },
+              {
+                as: "legalRepresentative",
+                model: User
+              }
+            ],
+            order: [["users", "createdAt", "DESC"]]
+          }),
+        genericErrorHandler
+      )
+    )
+    .chain(organizationModel =>
+      fromEither(
+        toOrganizationObject(organizationModel).map(organization => {
+          return ResponseSuccessCreation(organization);
```

Comment:
```suggestion
          return ResponseSuccessJson(organization);
```

ResponseSuccessCreation should be used without a payload (ResponseSuccessCreation
should return just the id with the Location header):
https://github.com/teamdigitale/io-functions/blob/master/api/public_api_v1.yaml#L324

Use ResponseSuccessJson if you want to return a payload